cmd/tpm-manager/tpm-manager.go

```diff
@@ -0,0 +1,506 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+Copyright 2015 CoreOS, Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/spf13/pflag"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/componentconfig"
+	"k8s.io/kubernetes/pkg/client/cache"
+	"k8s.io/kubernetes/pkg/client/leaderelection"
+	"k8s.io/kubernetes/pkg/client/record"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/controller"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/flag"
+	nodeutil "k8s.io/kubernetes/pkg/util/node"
+	"k8s.io/kubernetes/pkg/util/tpm"
+	"k8s.io/kubernetes/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+// Flag a node as untrusted
+func invalidateNode(node *api.Node) error {
+	if node.ObjectMeta.Annotations == nil {
+		node.ObjectMeta.Annotations = make(map[string]string)
+	}
+        newTaints := []api.Taint{}
+        untrustedTaint := api.Taint{
+                Key:    "Untrusted",
+                Value:  "true",
+                Effect: api.TaintEffectNoSchedule,
```

Comment:
I believe that this is the only thing implemented at the moment. It's a great question - a system will only transition to untrusted state if there's either

(a) a policy change or
(b) a reboot

In (a), the pod was already running in that configuration, so there's no additional harm in leaving it there. In (b) the containers will have died, but theoretically they could be restarted before we re-verify. So yeah, we do need to think about this, but I think the current situation is acceptable.